### PR TITLE
Fix ``XDeclaredButNotUsed`` warning when locks or rlocks module is used

### PR DESCRIPTION
**Pull request by vegansk**
Some parts of ``syslocks`` module is used in ``locks`` and some of them in ``rlocks`` module. ``syslocks`` is included in these modules. It leads to the compiler's warnings about unused types and procs. This PR fixes them.

### CODE REVIEW

**Comment by Araq** [comment]
Meh, is that really the way to fix these warnings? Why not use what's needed and disable what's unused?

**Comment by cheatfate** [comment]
I agree with @Araq, i think the biggest problem is in 
`include "system/syslocks"`

**Comment by vegansk** [comment]
Like this?

**Comment by Araq** [comment]
Declare the insideRLocksModule twice and use ``when insideRLocksModule`` please.

**Comment by vegansk** [comment]
@Araq, not twice but three times. ``syslocks`` is used in ``system``, ``locks`` and ``rlocks``. That's why I used ``when compiles`` before.

It will be cool if this thing will be written in some ``stdlib design`` document :-) The document that describes the design philosophy of the library, what to use and what not for example. This document describes only the style of stdlib: http://nim-lang.org/docs/nep1.html

File: lib/system.nim [hunk 1]
```diff
@@ -2561,6 +2561,16 @@ else:
     if x < 0: -x else: x
 {.pop.}
 
+proc compiles*(x: expr): bool {.magic: "Compiles", noSideEffect, compileTime.} =
```
Comment (Araq) [inline lib/system.nim]: Don't move 'compiles' around. Shouldn't be necessary anymore.

**Comment by cheatfate** [comment]
@vegansk your patch does not resolved [XDeclaredButNotUsed], 
Try to import both files and you will see even more [XDeclaredButNotUsed] warnings...

I think the best way to patch this warnings is to move all declarations from syslocks.nim to rlocks.nim...

**Comment by vegansk** [comment]
@cheatfate, @Araq, the problem is not in ``rlocks`` module. For example, there is ``waitSysCondWindows`` definition in ``syslocks``, that is only used in ``sysspawn.nim``, so here we go with ``locks``:
```
Hint: locks [Processing]
c:\apps\nimdev\lib\system\syslocks.nim(75, 10) Hint: 'locks.waitSysCondWindows(cond: var SysCond)' is declared but not used [XDeclaredButNotUsed]
```

And if we will need some SysLock, SysCond functions somewhere in stdlib, we will add it to syslocks, and here is the problem again.

So, I see two ways to solve this problem. The first way is to export definitions from syslocks and use ``import`` instead of ``include``. Don't know, to what it leads if it must be imported in system.nim. Araq, is it possible?

The second way is to forget about theese warnings and just switch them off for syslocks. I agree that disabling the warnings is a bad practice, but every tool has it's applications, and I think that this is the right place for it.

**Comment by Araq** [comment]
@vegansk system.nim cannot import modules. Well ok, it does import stuff but who know what kind of bugs it might trigger.

**Pull request merged**